### ISSUE TITLE
Separate `StarEncoder` class

### ISSUE THREAD

**Issue by ismael-mendoza** [prob-ml/bliss#697]
It's not clear to me why we should treat stars differently than galaxies/lenses and have their flux be estimated in the `DetectionEncoder`. It would make more sense (I think) to have a separate `StarEncoder` that also centers the star in their padded tile before estimating their flux. 

I suspect we would gain two benefits from this (1) The detection encoder will have an easier time with the optimization since it can now only focus on detections rather than having to juggle (somewhat arbitrarily) the star flux optimization (2) The flux estimation for stars will be improved, since stars will now always be centered making the problem easier for the new encoder. 

This more or less hinges on deprecating StarNet support (#691) which I think we are more or less agreeing that it would be a good idea?

**Comment by dereklhansen** [comment]
> (1) The detection encoder will have an easier time with the optimization since it can now only focus on detections rather than having to juggle (somewhat arbitrarily) the star flux optimization

With FAVI, I don't think this a big issue. In the loss function, the fluxes are optimized separately from the detections. There is some coupling in the architecture, since fluxes and locations come from the final layer of the same neural network.

**Comment by dereklhansen** [comment]
That said, I can see how it could be better to have a separate `StarEncoder`
